### ISSUE TITLE
rest_tornado: javascript client websocket connection failed

### ISSUE THREAD

**Issue by mahresohl** [saltstack/salt#34986]
###
I'm not able to test javascript client code provided as part of the rest_tornado documentation. I tried running it both in browser (Chrome, Firefox), and in NodeJS.

Firefox: The connection to ws://localhost:8000/ was interrupted while the page was loading.
Chrome: WebSocket connection to 'ws://localhost:8000/' failed: WebSocket is closed before the connection is established.

Note that the following works Fine:
- Using curl to send HTTP requests.
- Websocket client with Tornado (https://github.com/hiroakis/tornado-websocket-example) running in parallel on the same machine as Salt-api/rest_tornado

Also,
'salt-api -l debug' shows Get requests are being received when Websocket fails.

### Setup
Tested on Ubuntu 14 and 16 using Salt 2016.3.1 and 2016.3.0-960-g55bb3d0

/etc/salt/master:
<pre>
rest_tornado:
  port: 8000
  host: 0.0.0.0
  disable_ssl: true
  websockets: True
  cors_origin: '*'

external_auth:
  pam:
    userx:
      - .*
      - '@runner'
      - '@wheel'
</pre>

### Steps to Reproduce Issue
Just follow the instructions and sample Javascript provided with rest_tornado documentation.

### Versions Report
<pre>
Salt Version:
       Salt: 2016.3.0-960-g55bb3d0

Dependency Versions:
      cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
      gitdb: Not Installed
      gitpython: Not Installed
      ioflo: Not Installed
      Jinja2: 2.8
      libgit2: Not Installed
      libnacl: Not Installed
      M2Crypto: 0.21.1
      Mako: Not Installed
      msgpack-pure: Not Installed
      msgpack-python: 0.4.7
      mysql-python: Not Installed
      pycparser: Not Installed
      pycrypto: 2.6.1
      pygit2: Not Installed
      Python: 2.7.12 (default, Jul  1 2016, 15:12:24)
      python-gnupg: Not Installed
      PyYAML: 3.11
      PyZMQ: 15.3.0
      RAET: Not Installed
      smmap: Not Installed
      timelib: Not Installed
     Tornado: 4.4.1
     ZMQ: 4.1.5

System Versions:
      dist: Ubuntu 16.04 xenial
      machine: x86_64
      release: 4.4.0-31-generic
      system: Linux
      version: Ubuntu 16.04 xenial
</pre>

**Comment by Ch3LL** [comment]
@mahresohl looks like there is a syntax error in the docs which I ran into when trying to recreate this. `Websocket` should be `WebSocket` in the javascript. 

Although once I got that setup I am running into the same error as you. I'm not javascript savy so i'm not sure what the fix on this would be to make sure the example is accurate. 

ping @jacksontj looks like you wrote these docs. Is there something we are missing on the configs to get this to work? Maybe something that needs to be changed in the docs. Thanks

**Comment by jacksontj** [comment]
I actually wasn't involved in the websockets API. I just wrapped it in the
config option (as to not load it by default) since the websockets code
currently has no test coverage.

**Comment by mahresohl** [comment]
It seems to be a server-side problem, not the client. Server gets the requests but drops the connection for some reason.

**Comment by HWDexperte** [comment]
The described problem here is due to the last line in the docs

```javascript
// Terminates websocket connection and Salt's "real time" event stream on the server.
source.close();
```

This is interrupting the WebSocket connection.

But i still have a problem, as i try to pass the generated token as described to the url, the servers response is `403 Forbidden`. Some help needed

**Comment by mahresohl** [comment]
That's true indeed. After I removed source.close() I see a different error in the salt-api log
----------------
<pre>
...
403 GET /all_events/
Cross origin websockets not allowed
</pre>
-------------------
Even though /etc/salt/master uses "cors_origin: '*'". 
But that's a different bug all together.

**Comment by cprior** [comment]
I came here by searching the issues for "tornada".

Caveat: If you actually do NOT close the websocket, you will end up with a very big JSON as each job gets appended to the data structure.
